src/main/java/uk/gov/hmcts/ccd/domain/service/search/CaseSearchResultViewGenerator.java

```diff
@@ -61,25 +64,78 @@ public CaseSearchResultView execute(String caseTypeId,
 
     private boolean itemsRequireFormatting(List<SearchResultViewHeaderGroup> headerGroups) {
         return headerGroups.size() == 1
-               && headerGroups.get(0).getFields().stream().anyMatch(field -> field.getDisplayContextParameter() != null);
+            && headerGroups.get(0).getFields().stream().anyMatch(field -> field.getDisplayContextParameter() != null);
     }
 
     private List<SearchResultViewItem> buildItems(String useCase, CaseSearchResult caseSearchResult, String caseTypeId, List<String> requestedFields) {
         CaseTypeDefinition caseTypeDefinition = getCaseTypeDefinition(caseTypeId);
         SearchResult searchResultDefinition = searchResultDefinitionService.getSearchResultDefinition(caseTypeDefinition, useCase, requestedFields);
 
         List<SearchResultViewItem> items = new ArrayList<>();
-        // Only one case type is currently supported so we can reuse the same definitions for building all items
-        caseSearchResult.getCases().forEach(caseDetails ->
-            items.add(buildSearchResultViewItem(caseDetails, caseTypeDefinition, searchResultDefinition)));
-
+        caseSearchResult.getCases().forEach(caseDetails -> {
+            filterResultsByAuthorisationAndUserRole(useCase, caseDetails, caseTypeId, requestedFields);
+            items.add(buildSearchResultViewItem(caseDetails, caseTypeDefinition, searchResultDefinition));
+        });
         return items;
     }
 
-    private List<SearchResultViewHeaderGroup> buildHeaders(String caseTypeId,
-                                                           String useCase,
-                                                           CaseSearchResult caseSearchResult,
-                                                           List<String> requestedFields) {
+    private CaseDetails filterResultsByAuthorisationAndUserRole(String useCase, CaseDetails caseDetails, String caseTypeId, List<String> requestedFields) {
+        CaseTypeDefinition caseTypeDefinition = getCaseTypeDefinition(caseTypeId);
+        caseDetails.getData().entrySet().removeIf(
+            caseField -> {
+                Optional<CaseFieldDefinition> caseFieldDefinition = caseTypeDefinition.getCaseField(caseField.getKey());
```

Comment:
minor: since the field def is only used by filterResultsByAuthorisationAccessOnField you can then move it into that method itself?